realm/src/main/java/io/realm/Realm.java

```diff
@@ -421,13 +430,23 @@ private static Realm create(File writableFolder, String filename, byte[] key) {
         }
     }
 
-    private static Realm createAndValidate(String absolutePath, byte[] key, boolean validateSchema, boolean autoRefresh) {
+    private static synchronized Realm createAndValidate(String absolutePath, byte[] key, boolean validateSchema,
```

Comment:
Do you need to sync the whole method? I prefer only the code block needed for clarity of what needs synced.